### PR TITLE
Allow resolution adaption of the captured screen

### PR DESCRIPTION
**Pull request by gubjack**
It might be useful the adapt the resolution of a captured screen as one would be able with a normal camera.

### CODE REVIEW

File: webcam-capture-drivers/driver-screencapture/src/main/java/com/github/sarxos/webcam/ds/gstreamer/ScreenCaptureDevice.java [hunk 1]
```diff
@@ -59,14 +62,25 @@ public Dimension getResolution() {
...
 
 	@Override
 	public BufferedImage getImage() {
 		final GraphicsConfiguration gc = device.getDefaultConfiguration();
 		final Rectangle bounds = gc.getBounds();
-		return robot.createScreenCapture(bounds);
+		BufferedImage screen = robot.createScreenCapture(bounds);
+		int width = resolution.width;
+		int height = resolution.height;
+		BufferedImage img = new BufferedImage(width, height, screen.getType());
```
Comment (sarxos) [inline webcam-capture-drivers/driver-screencapture/src/main/java/com/github/sarxos/webcam/ds/gstreamer/ScreenCaptureDevice.java]: Please consider checking if screen image from Robot has the same size as resolution Dimension and return original screen image in such a case, so we do not need to create resized image when it's unnecessary.

```java
if (screen.getWidth() == width && screen.getHeight() == height) {
  return screen;
}
```

**Comment by sarxos** [comment]
Hi @gubjack, thank you for this PR :) This is really nice functionality. I added one comment in regards to the changes you made.

**Comment by gubjack** [comment]
You’re welcome!

I’ll take care the next days.  To much other work …

Hagen

**Comment by sarxos** [comment]
Sure. I can merge this now and add mentioned fix later if you like.

**Comment by sarxos** [comment]
Thank you @gubjack! I'm merging this.

**Comment by gubjack** [comment]
:OK:

**Pull request merged**